pkg/networkserver/grpc_ns_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkserver_test
+
+import (
+	"testing"
+
+	"github.com/smartystreets/assertions"
+	"go.thethings.network/lorawan-stack/pkg/component"
+	. "go.thethings.network/lorawan-stack/pkg/networkserver"
+	"go.thethings.network/lorawan-stack/pkg/types"
+	"go.thethings.network/lorawan-stack/pkg/util/test"
+	"go.thethings.network/lorawan-stack/pkg/util/test/assertions/should"
+)
+
+func TestGenerateDevAddrAssignment(t *testing.T) {
+	a := assertions.New(t)
+
+	// Use DevAddr prefix from NetID.
+	{
+		ns := test.Must(New(
+			component.MustNew(test.GetLogger(t), &component.Config{}),
+			&Config{
+				NetID:               types.NetID{0x00, 0x00, 0x13},
+				DeduplicationWindow: 42,
+				CooldownWindow:      42,
+				DownlinkTasks: &MockDownlinkTaskQueue{
+					PopFunc: DownlinkTaskPopBlockFunc,
+				},
+			})).(*NetworkServer)
+
+		devAddr, _ := ns.GenerateDevAddrAssignment(test.Context(), nil)
```

Comment:
Call this via the gRPC client(https://github.com/TheThingsNetwork/lorawan-stack/pull/758/files#diff-e699b03edbcaf36d0a1bd906c75d1d22R584)
Look at the registry tests for an example on how to do that.